datalad/support/sshconnector.py

```diff
@@ -49,14 +50,18 @@ def __init__(self, ctrl_path, host):
 
         host: str
           host to connect to. This may include the user ( [user@]host )
+
+        port: str
+          port to connect over
         """
         self._runner = None
 
         # TODO: This may actually also contain "user@host".
         #       So, better name instead of 'host'?
         self.host = host
-        self.ctrl_path = ctrl_path
-        self.cmd_prefix = ["ssh", "-S", self.ctrl_path, self.host]
+        self.ctrl_path = ctrl_path + ":" + port if port else ctrl_path
+        self.port = port
+        self.ctrl_options = ["-o", "ControlPath=\'" + self.ctrl_path + "\'"]
```

Comment:
On Tue, 12 Jul 2016, Debanjum wrote:
>  > +        self.ctrl_options = ["-o", "ControlPath=\'" + self.ctrl_path + "\'"]

>    wouldn't this deal with escaping weird characters in the self.ctrl_path
>    string ?

which consists of the path, server name and a port.. so what weird characters
we are talking about? ;)  actually -- please test.  Use
get_most_obscure_supported_name() to generate some very nice path to the
control path and see if it works out? ;)

but I think that is the point of using a list there since then no
escaping would be needed (I think, would need to double check since
shell=True there... may be we don't need that anyways for scp)

-- 
Yaroslav O. Halchenko
Center for Open Neuroscience     http://centerforopenneuroscience.org
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik